src/Simulation/Core/Submitters/IQSharpSubmitter.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using Microsoft.Quantum.Simulation.Core;
+using System.Threading.Tasks;
+
+namespace Microsoft.Quantum.Runtime.Submitters
+{
+    /// <summary>
+    /// An interface for submitting Q# programs to Azure.
+    /// </summary>
+    public interface IQSharpSubmitter : IAzureSubmitter
+    {
+        /// <summary>
+        /// Submits a job to execute a Q# program without waiting for execution to complete.
+        /// </summary>
+        /// <typeparam name="TIn">The entry point argument type.</typeparam>
+        /// <typeparam name="TOut">The entry point return type.</typeparam>
+        /// <param name="entryPoint">The entry point information for the submitted program.</param>
+        /// <param name="argument">The argument to the entry point.</param>
+        /// <param name="options">Additional options for the submission.</param>
+        /// <returns>The submitted job.</returns>
+        Task<IQuantumMachineJob> SubmitAsync<TIn, TOut>(
```

Comment:
I think this interface should also provide an option without the submission options.

Reply:
What is the behavior when the options are not provided? Is it specific to each submitter instance, or are there defaults that can be used across all submitters?

I think it would also be reasonable to allow e.g. the default friendly name, but a custom number of shots, or vice versa. So we could have a default instance of SubmissionOptions as a static property there, which would make the submitter APIs something like:

```csharp
submitter.SubmitAsync(entryPoint, argument, SubmissionOptions.Default);
submitter.SubmitAsync(entryPoint, argument, SubmissionOptions.Default.With(shots: 1000));
submitter.SubmitAsync(entryPoint, argument, SubmissionOptions.Default.With(friendlyName: "foo", shots: 1000));
```

(This would work better with a record, but we don't have C# 9 yet.)

Reply:
If necessary, we could provide a `submitter.SubmitAsync(entryPoint, argument)` that automatically used `SubmissionOptions.Default`. But I am wary of overloads and default parameters, so I would prefer to keep `SubmissionOptions.Default` explicit in the calling code.

Reply:
I added a default instance for SubmissionOptions, where the friendly name is empty string, and the number of shots is 500.

Reply:
I really liked your idea of a default instance for `SubmissionOptions`. It makes the interface simpler too because we don't have a bunch of overloads.